com/bl/ttt/tictactoe.py

```diff
@@ -49,15 +50,23 @@ def get_previous_turn_player(self):
         else:
             return self.player1
 
-    def get_player_move_log(self):
-        return self.player_move_log
+    def get_player_move_log(self, player_id='both'):
+        if player_id == 'both':
+            x_move_log = list(self.player_move_log.get('X'))
+            o_move_log = list(self.player_move_log.get('O'))
+            players_move_tuples = list(itertools.zip_longest(x_move_log, o_move_log))
+            players_move_list = list(itertools.chain(*players_move_tuples))
+            if len(players_move_list) == 0:
+                return []
```

Comment:
No need to return [] here the result is already [] :)

Reply:
Understood: it is now with your fine list comprehension.